project-docs/stumptown-writers-guide.md

```diff
@@ -0,0 +1,697 @@
+    - title: Reference
+      pages:
+          - "/content/html/reference/elements"
+---
+
+```
+
+The `title`, `mdn_url`, `specifications`, and `recipe` properties are all the same as in a normal recipe-driven page like `html-element`. The `related_content` is in the landing page itself rather than in the recipe: this is because different landing pages want different sidebars, while all the HTML elements should get the same sidebar.
+
+The other property is a `link_lists` object describing the lists of links to include.
+
+The Markdown part of the landing page provides an overview of the technology.
+
+## Guide pages
+
+For reference pages (and to a lesser extent landing pages) we've described a system where a page has a clear structure that's described in the recipe and enforced by stumptown.
+
+I'd estimate that about 90% of the web documentation pages on MDN are reference pages and will be (we hope) amenable to this treatment.
+
+But MDN also contains a substantial number of "guide" pages. These pages are fundamentally unstructured prose content, and the recipe model doesn't work for them. Still, we can't treat them just as prose, because they want to embed special content such as browser compatibility tables or live samples - and they want to be able to embed them just anywhere.
+
+In stumptown, guide pages are Markdown files with YAML front matter. The front matter is quite minimal:
+
+```yaml
+---
+title: Applying color to HTML elements using CSS
+mdn_url: /en-US/docs/Web/HTML/Applying_color
+related_content: /content/related_content/html.yaml
+recipe: guide
+---
+
+```
+
+Like everything else, it has a `title` and a `mdn_url` and a `recipe`. Like landing pages, it has `related_content` in the front matter rather than the recipe, because because different guide pages want different sidebars. These are the only items allowed in the front matter.
+
+For guide pages, the recipe is empty, so doesn't put any constraints on the things that the page can contain.
+
+The Markdown part is of course the guide's prose. To support the "special content" that guides want to embed, stumptown has the concept of "directives". A directive looks like:
+
+```
+{{{{{embed-example:examples/specifying-colors-in-stylesheets}}}}}
+```
+
+We currently support two directives:
+
+* "embed-example": this tells stumptown that the guide wants to embed a live sample at this point. Its argument is "examples/specifying-colors-in-stylesheets", which is a path to a directory containing an example. The content of this directory is just the same as the directories references by the `examples` front matter property, with a mandatory "description.md" and optional "example.[html|css|js]" files.
+
+* "embed-compat": this tells stumptown that the guide wants to embed a BCD table at this point. Its argument is a BCD query string like "html.elements.audio".
+
+We will probably add support for more directives in future, not too many. One likely candidate would be "embed-link-list".
```

Comment:
+1 for `embed-link-list`.

Do we need an `embed-media` directive to embed a video or audio object in the page? Possibly also `embed-youtube` or such. I ask because there's nothing in this document about how to embed video.

Reply:
How many pages use this? In MDN we have 76 pages (en-US) that use EmbedYoutube. They are almost all in devtools, which won't be in stumptown. So I don't see a current need for this.